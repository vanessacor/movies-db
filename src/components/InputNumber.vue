<template>
	<div>
		<label for="title">{{ label }}</label>
		<input
			class="input"
			v-bind="$attrs"
			v-bind:value="value"
			v-on:input="$emit('input', $event.target.value)"
		/>
	</div>
</template>

<script>
	export default {
		name: "inputNumber",
		inheritAttrs: false,
		props: ["label", "value"],
	};
</script>