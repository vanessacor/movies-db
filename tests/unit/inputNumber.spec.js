import { mount } from "@vue/test-utils";
import InputNumber from "@/components/InputNumber";

describe("test InputText.vue", () => {
  let wrapper = null;
  const value = "4";
  const label = "Rate the movie";
  beforeEach(() => {
    wrapper = mount(InputNumber, {
      propsData: { value, label },
    });
  });
  afterEach(() => {
    wrapper.destroy();
  });
  it("renders props when passed", () => {
    expect(wrapper.vm.value).toMatch(value);
    expect(wrapper.vm.label).toMatch(label);
  });

  it("emits an event", () => {
    wrapper.vm.$emit("rating");
    expect(wrapper.emitted().rating).toBeTruthy();
  });
});
