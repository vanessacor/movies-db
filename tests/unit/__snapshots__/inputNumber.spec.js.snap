// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test InputText.vue match the snapshot 1`] = `<div><label for="title">Rate the movie</label> <input class="input"></div>`;
